clifford/__init__.py

```diff
@@ -258,17 +260,77 @@ def general_exp(x, max_order=15):
     for i in range(1, max_order):
         if np.any(np.abs(tmp.value) > _settings._eps):
             tmp = tmp*scaled * (1.0 / i)
-            result += tmp
+            result = result + tmp
         else:
             break
 
     # undo scaling
     while scale > 1:
-        result *= result
+        result = result*result
         scale >>= 1
     return result
 
 
+@_numba_utils.njit
+def general_sin(X, max_order=30):
+    """
+    A taylor series expansion for sin
+    """
+    op = 0 * X
+    for n in range(max_order):
+        op = op + ((-1) ** (n) / math.gamma(2 * n + 2)) * X ** (2 * n + 1)
+    return op
+
+
+@_numba_utils.njit
+def general_cos(X, max_order=30):
+    """
+    A taylor series expansion for cos
+    """
+    op = 0 * X
+    for n in range(max_order):
+        op = op + ((-1) ** (n) / math.gamma(2 * n + 1)) * X ** (2 * n)
+    return op
+
+
+def general_tan(X, max_order=30):
+    """
+    The tan function as the ratio of sin and cos
+    Note. It would probably be better to implement this as its own taylor series.
+    """
+    return general_sin(X, max_order) / general_cos(X, max_order)
+
+
+@_numba_utils.njit
+def general_sinh(X, max_order=30):
+    """
+    A taylor series expansion for sinh
+    """
+    op = 0 * X
+    for n in range(max_order):
+        op = op + (1 / math.gamma(2 * n + 2)) * X ** (2 * n + 1)
+    return op
+
+
+@_numba_utils.njit
+def general_cosh(X, max_order=30):
+    """
+    A taylor series expansion for cosh
+    """
+    op = 0 * X
+    for n in range(max_order):
+        op = op + (1 / math.gamma(2 * n + 1)) * X ** (2 * n)
+    return op
+
+
+def general_tanh(X, max_order=30):
+    """
+    The tanh function as the ratio of sinh and cosh
+    Note. It would probably be better to implement this as its own taylor series.
+    """
+    return general_sinh(X, max_order) / general_cosh(X, max_order)
```

Comment:
Should we put these in a new `taylor_expansions` submodule?

Reply:
I think that is probably a good idea